### PR TITLE
You can now unextend and extend the RL-72.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Title.
It starts unextended, and will be normal sized, but cannot fire.
Extending it will make you able to fire, but will make it bulky.

## Why It's Good For The Game
Makes it a bit more unique overall. I think I could make a disposable RPG that can't be extended under any circumstances in another PR, maybe.

## Changelog
:cl:
balance: You can extend and unextend the RL-72 one use RPG, which will make it storeable when retracted, and fireable when extended.
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Neat

**Comment by TiviPlus** [comment]
Whats a t72

**Review by Kuro020 — APPROVED** [review]

**Comment by Kuro020** [comment]
> Whats a t72

a gun

File: code/modules/projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -711,18 +711,20 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
...
 	max_shells = 1 //codex
 	caliber = CALIBER_84MM //codex
 	load_method = SINGLE_CASING //codex
+	w_class = WEIGHT_CLASS_NORMAL
 	default_ammo_type = /obj/item/ammo_magazine/rocket/oneuse
 	allowed_ammo_types = list(/obj/item/ammo_magazine/rocket/oneuse)
 	reciever_flags = AMMO_RECIEVER_CLOSED|AMMO_RECIEVER_MAGAZINES
 	flags_equip_slot = ITEM_SLOT_BELT
 	attachable_allowed = list(/obj/item/attachable/magnetic_harness)
+	var/extended = 0
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:727]: Use bools properly

File: code/modules/projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -732,6 +734,23 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
 	recoil = 3
 	scatter = -100
 
+/// Do a short windup, swap the extension status of the rocket if successful, then swap the flags.
+/obj/item/weapon/gun/launcher/rocket/oneuse/unique_action(mob/living/user)
+	playsound(user, 'sound/weapons/guns/misc/oneuse_deploy.ogg', 25, 1)
+	if(do_after(user, 20, TRUE, src, BUSY_ICON_DANGER))
+		extended = !extended
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:741]: Early return

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by TiviPlus** [comment]
I said early return the do after not the extended

**Pull request merged**